File: Vanilla/sources/zomboid/zombie/network/DBTicket.java
package zombie.network;

public class DBTicket {
   private String author = null;
   private String message = "";
   private int ticketID = 0;
   private boolean viewed = false;
   private DBTicket answer = null;
   private boolean isAnswer = false;

   public DBTicket(String var1, String var2, int var3) {
      this.author = var1;
      this.message = var2;
      this.ticketID = var3;
      this.viewed = this.viewed;
   }

   public String getAuthor() {
      return this.author;
   }

   public void setAuthor(String var1) {
      this.author = var1;
   }

   public String getMessage() {
      return this.message;
   }

   public void setMessage(String var1) {
      this.message = var1;
   }

   public int getTicketID() {
      return this.ticketID;
   }

   public void setTicketID(int var1) {
      this.ticketID = var1;
   }

   public boolean isViewed() {
      return this.viewed;
   }

   public void setViewed(boolean var1) {
      this.viewed = var1;
   }

   public DBTicket getAnswer() {
      return this.answer;
   }

   public void setAnswer(DBTicket var1) {
      this.answer = var1;
   }

   public boolean isAnswer() {
      return this.isAnswer;
   }

   public void setIsAnswer(boolean var1) {
      this.isAnswer = var1;
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/DBTicket.lua
---@class DBTicket : zombie.network.DBTicket
---@field private author String
---@field private message String
---@field private ticketID int
---@field private viewed boolean
---@field private answer DBTicket
---@field private isAnswer boolean
DBTicket = {}

---@public
---@param arg0 boolean
---@return void
function DBTicket:setViewed(arg0) end

---@public
---@return int
function DBTicket:getTicketID() end

---@public
---@return boolean
function DBTicket:isViewed() end

---@public
---@param arg0 boolean
---@return void
function DBTicket:setIsAnswer(arg0) end

---@public
---@param arg0 int
---@return void
function DBTicket:setTicketID(arg0) end

---@public
---@return String
function DBTicket:getAuthor() end

---@public
---@return DBTicket
function DBTicket:getAnswer() end

---@public
---@return boolean
function DBTicket:isAnswer() end

---@public
---@return String
function DBTicket:getMessage() end

---@public
---@param arg0 DBTicket
---@return void
function DBTicket:setAnswer(arg0) end

---@public
---@param arg0 String
---@return void
function DBTicket:setMessage(arg0) end

---@public
---@param arg0 String
---@return void
function DBTicket:setAuthor(arg0) end
